gips/data/sentinel2/sentinel2.py

```diff
@@ -964,26 +964,36 @@ def fmask_geoimage(self, asset_type):
                 metadata_xml = "{}/{}".format(self._temp_proc_dir, name)
         if not metadata_xml:
             raise IOError("{} does not contain expected metadatafile.".format(safe_zip.filename))
-        subprocess.check_call(
-            [
-                "fmask_sentinel2makeAnglesImage.py",
-                "-i", metadata_xml,
-                "-o", "%s/angles.img" % self._temp_proc_dir,
-            ]
-            #stderr=DEVNULL
-        )
         safe_zip.close()
 
-        subprocess.check_call(
-            [
-                "fmask_sentinel2Stacked.py",
-                "-a", "%s/allbands.vrt" % self._temp_proc_dir,
-                "-z", "%s/angles.img" % self._temp_proc_dir,
-                "-o", "%s/cloudmask.tif" % self._temp_proc_dir,
-                "-v",
-            ],
-            stderr=DEVNULL
-        )
+        angles_cmd_list = [
+            "fmask_sentinel2makeAnglesImage.py",
+            "-i", metadata_xml,
+            "-o", "%s/angles.img" % self._temp_proc_dir,
+        ]
+        fmask_cmd_list = [
+            "fmask_sentinel2Stacked.py",
+            "-a", "%s/allbands.vrt" % self._temp_proc_dir,
+            "-z", "%s/angles.img" % self._temp_proc_dir,
+            "-o", "%s/cloudmask.tif" % self._temp_proc_dir,
+            "-v",
+        ]
+        # Temp dir for intermediaries.  mask is output to self._temp_proc_dir.
+        with utils.make_temp_dir(prefix='gips-py-fmask', dir='/tmp') as tdir:
+            utils.verbose_out('running: ' + ' '.join(angles_cmd_list), 3)
+            prev_wd = os.getcwd()
+            os.chdir(tdir)
+            subprocess.check_call(
+                angles_cmd_list,
+                #stderr=DEVNULL
+            )
+
+            utils.verbose_out('running: ' + ' '.join(fmask_cmd_list), 3)
+            subprocess.check_call(
+                fmask_cmd_list,
+                stderr=DEVNULL
+            )
+            os.chdir(prev_wd)
```

Comment:
Optional:  I can't decide if there's a significant risk to letting the working directory remain changed in the case of an uncaught exception before this line is reached.  I think chdir is a bit harmless as gips works with absolute paths for everything, so it may be okay to leave it.

Reply:
The `chdir` is because 3rd party stuff uses the working directory....and if everything goes well, I don't want to be hanging onto that directory in some way, such that you couldn't unmount a filesystem.  I suppose adding the try-finally might be best.

Reply:
Good point about the tempdir hanging around, I hadn't thought of that.  Also, what do you think about a comment explaining the need for the chdir, while you're in there?  Just for clarity's sake.